### DISCUSSION TITLE
Center pointy_rectangle to the window and field_of_movement input

### DISCUSSION THREAD

**Discussion by Godnoken** [ManevilleF/hexx#134] [Q&A] [answered]
Hi,

I have two issues;

The first one is that I am struggling to center a 15 by 11 sized pointy_rectangle.

The only way I manage to do so at the moment is by doing something like ``shapes::pointy_rectangle([-7, 7, -5, 5])``.
It does center it, however, the x, y coordinates I receive are not desirable in that solution. I don't want any negative x, y, but 0-14 and 0-10 instead.

Is there an easy way to solve this? I assume it would involve not using ``hex_to_world_pos``.

Edit;

I 'solved' the first problem by continuing to use ``[-7, 7, -5, 5]``

with

````
hex.x = (hex.x - 7).abs();
hex.y = (hex.y - 5).abs();
````

and inverting x, y on the layout so I can present 0,0 in the top left corner of the rectangle and 14, 10 in the bottom right.

Of course, this is purely to visualize the tile's x, y with text, and I continue to use the ``[-7, 7, -5, 5]`` coordinates for the position.

Not ideal but it works for my purposes.

---

The other issue is that the visual placement is for some reason not the same as the coords that ``handle_input`` picks up (taken from "field_of_movement" example). It's off by 1 tile, as if there's another outer layer of hexagons. Maybe that's by design from the example and I just don't see where?

---

Thank you so much for this amazing crate! Let me know if I need to clarify anything.

**Comment by ManevilleF** [comment, marked as answer]
You could change the world origin in the `HexLayout` to offset everything, but if you are using the `shapes` module you might need to rewrite the rectangle function to do what you intend and offset all hexagons to avoid negative values, this combined with the `origin` change would make a centered rectangle

**Reply by Godnoken** [reply]
It seems like ``origin`` is broken, or I've got it all set up the wrong way. This is what it looks like with ``origin: Vec2::new(0.0, 150.0)``;


https://github.com/ManevilleF/hexx/assets/17499594/b5cfc6ae-6ada-48c0-89f6-6f9a35301ef6

**Reply by Godnoken** [reply]
Interestingly, ``origin: Vec2::new(150.0, 0.0)`` actually does offset the x-axis, although having the same issues with mouse displacement etcetera. Changing the y-axis only displaces & glitches.

**Reply by ManevilleF** [reply]
Just by changing the origin you get this ? looks like a bug

**Reply by Godnoken** [reply]
Yup. Try the example given in ``field_of_movement`` and change the origin and you'll get this result.

**Reply by ManevilleF** [reply]
I think know what the problem is, the `PlaneMeshBuilder` that creates the hex mesh takes in account the layout origin, and shifts everything

**Reply by ManevilleF** [reply]
<img width="1478" alt="Screenshot 2024-01-12 at 14 24 50" src="https://github.com/ManevilleF/hexx/assets/26703856/e960f313-bd2c-4819-9f7f-c4c3c2e92761">

Fixes it, I gotta rethink that API, because it makes sens to take in accout the origin when generating meshes, unless, like for bevy we want the mesh vertices to be centered around `0, 0`

**Reply by Godnoken** [reply]
Brilliant, thank you very much! :)

**Reply by ManevilleF** [reply]
Could you check #139 @Godnoken ? The new `center_aligned` method should fix that problem

**Comment by Godnoken** [comment]
Thank you! I will look into it.
